amethyst_assets/src/asset.rs

```diff
@@ -155,15 +166,15 @@ where
     /// The kind of error it may produce.
     type Error: Error;
 
-    /// Returns a list of supported extension (without `.`).
+    /// Returns a list of media types with supported extensions (without `.`).
     ///
     /// ## Examples
     ///
-    /// * `"png"`
-    /// * `"obj"`
-    /// * `"wav"`
-    fn extensions() -> &'static [&'static str];
+    /// * `[("image/png", ["png"), ("image/jpeg", ["jpg", "jpeg"])]`
+    /// * `[("text/plain", ["obj"])]`
+    /// * `[("audio/wav", ["wav"])]`
+    fn media_extensions() -> &'static [(&'static MediaType, &'static [&'static str])];
```

Comment:
I'm in favor of the current approach. I just think it's not necessary to do this automatically.